opentelemetry-sdk/benches/log_processor.rs

```diff
@@ -0,0 +1,171 @@
+
+impl LogProcessor for NoopProcessor {
+    fn emit(&self, _data: &mut LogData) {}
+
+    fn force_flush(&self) -> LogResult<()> {
+        Ok(())
+    }
+
+    fn shutdown(&self) -> LogResult<()> {
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct CloningProcessor;
+
+impl LogProcessor for CloningProcessor {
+    fn emit(&self, data: &mut LogData) {
+        let _data_cloned = data.clone();
+    }
+
+    fn force_flush(&self) -> LogResult<()> {
+        Ok(())
+    }
+
+    fn shutdown(&self) -> LogResult<()> {
+        Ok(())
+    }
+
+}
+
+
+#[derive(Debug)]
+struct SendToChannelProcessor {
+    sender: std::sync::mpsc::Sender<LogData>,
+    receiver: Arc<Mutex<std::sync::mpsc::Receiver<LogData>>>,
+}
+
+impl SendToChannelProcessor {
+    fn new() -> Self {
+        let (sender, receiver) = std::sync::mpsc::channel();
+        let s = Self {             
+             sender : sender,
+             receiver : Arc::new(Mutex::new(receiver))
+            };
+        let receiver_cloned = s.receiver.clone();
+        let _ = std::thread::spawn(move || {
+            loop {
+                sleep(std::time::Duration::from_millis(10));
+                let data = receiver_cloned.lock().unwrap().recv();
+                if data.is_err() {
+                    println!("Error receiving log data from channel {0}", data.err().unwrap());
+                    break;
+                }
+            }
+        });
+        s
+    }
+}
+
+impl LogProcessor for SendToChannelProcessor {
+    fn emit(&self, data: &mut LogData) {
+        let data_cloned = data.clone();
+        let res =  self.sender.send(data_cloned);
+        if res.is_err() {
+            println!("Error sending log data to channel {0}", res.err().unwrap());
+        }
+    }
+
+    fn force_flush(&self) -> LogResult<()> {
+        Ok(())
+    }
+
+    fn shutdown(&self) -> LogResult<()> {
+        Ok(())
+    }
+}
+
+fn criterion_benchmark(c: &mut Criterion) {
+    log_noop_processor(c);
+    log_cloning_processor(c);
+    log_cloning_and_send_to_channel_processor(c);
+}
+
+fn log_noop_processor(c: &mut Criterion) {
+    let provider = LoggerProvider::builder()
+        .with_log_processor(NoopProcessor {})
+        .build();
+    let logger = provider.logger("benchmark");
+
+    c.bench_function("log_noop_processor", |b| {
+        b.iter(|| {
+            let mut log_record = logger.create_log_record();
+            let now = SystemTime::now();
+            log_record.set_observed_timestamp(now);
+            log_record.set_target("my-target".to_string());
+            log_record.set_event_name("CheckoutFailed");
+            log_record.set_severity_number(Severity::Warn);
+            log_record.set_severity_text("WARN");
+            log_record.add_attribute("book_id", "12345");
+            log_record.add_attribute("book_title", "Rust Programming Adventures");
+            log_record.add_attribute("message", "Unable to process checkout.");
+
+            logger.emit(log_record);
```

Comment:
nit - probably reuse the function:
```rust
fn create_log_record(logger: &Logger) -> LogData {
    let mut log_record = logger.create_log_record();
    let now = SystemTime::now();
    log_record.set_observed_timestamp(now);
    log_record.set_target("my-target".to_string());
    log_record.set_event_name("CheckoutFailed");
    log_record.set_severity_number(Severity::Warn);
    log_record.set_severity_text("WARN");
    log_record.add_attribute("book_id", "12345");
    log_record.add_attribute("book_title", "Rust Programming Adventures");
    log_record.add_attribute("message", "Unable to process checkout.");
    log_record
}
```
This will only have an ownership transfer of LogRecord, with stack-allocated data within it (if any). If possible,. else can be done in separate PR.